pgxn/neon/pagestore_smgr.c

```diff
@@ -2846,35 +2844,20 @@ slru_kind_from_string(const char* str, NeonSlruKind* kind)
 	return false;
 }
 
+#define BlockNumberToRegion(blkno) (blkno % MAX_REGIONS)
+
 /**
- * neon_slru_kind_check() - Check if the SLRU kind is supported by the pageserver
+ * neon_slru_is_remote_page() - Check if the SLRU kind is supported by the pageserver
  */
 bool
-neon_slru_kind_check(SlruCtl ctl)
+neon_slru_is_remote_page(SlruCtl ctl, BlockNumber blkno)
 {
 	const char *dir = ctl->Dir;
+	int region = BlockNumberToRegion(blkno);
 
-	if (strcmp(dir, "pg_xact") == 0 && neon_slru_clog)
-	{
-		return true;
-	}
-
-	if ((strcmp(dir, "pg_multixact/members") == 0 || strcmp(dir, "pg_multixact/offsets") == 0) &&
-		neon_slru_multixact)
-	{
-		return true;
-	}
-
-	if (strcmp(dir, "pg_csn") == 0 && neon_slru_csnlog) 
-	{
-		return true;
-	}
-
-	return false;
+	return neon_slru_csnlog && RegionIsRemote(region) && strcmp(dir, "pg_csn") == 0;
```

Comment:
Previously we were returning true for pg_multixact and pg_xact as well. So shouldn't we read the remote slrus from other regions? Or was this some bug that we didn't come across earlier?

Reply:
the code for reading remote slru only worked with the csn log because it assumes the pages of the different regions are interleaved, while other SLRUs does not follow this layout. There is no point in reading pg_multixact and pg_xact anyway because their values are local to the current postgres instance.